### PR TITLE
[Enhancement](HttpServer) Provide authentication interface for BE

### LINKED ISSUE #15658: [Enhancement] add http interface authentication and support https
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Currently, doris supports access to fe / be only through http protocol, not https protocol. In addition, http does not perform interface authentication, which may cause security problems.

### Solution

Add http interface authentication and support https for doris, eliminate potential security risks and strengthen enterprise-level security support.
[Support https protocol and interface authentication](https://cwiki.apache.org/confluence/display/DORIS/DSIP-029%3A+Support+https+protocol+and+interface+authentication)

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)]

### PR DESCRIPTION
**Pull request by yongjinhou**
# Proposed changes

Issue Number: close #15658

## Problem summary

1. Organize http documents
2. Add http interface authentication for FE
3. Support https interface for FE
4. **Provide authentication interface**
5. Add http interface authentication for BE
6. Support https interface for BE

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yongjinhou** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.16 seconds
 stream load tsv:          471 seconds loaded 74807831229 Bytes, about 151 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          68 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230223102659_clickbench_pr_102551.html

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run feut

**Comment by yongjinhou** [comment]
run clickbench

**Comment by yongjinhou** [comment]
run buildall

File: gensrc/thrift/FrontendService.thrift [hunk 1]
```diff
@@ -755,6 +755,38 @@ struct TConfirmUnusedRemoteFilesResult {
...
+  LOAD = 4,
+  ALTER = 5,
+  USAGE = 6,
+  CREATE = 7,
+  ALL = 8,
+  OPERATOR = 9,
+  DROP = 10
+}
+
+struct THttpAuthRequest {
```
Comment (yangzhg) [inline gensrc/thrift/FrontendService.thrift:772]: why called HttpAut

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run feut

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run compile

**Comment by yongjinhou** [comment]
run p1

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run buildall

File: gensrc/thrift/FrontendService.thrift [hunk 2]
```diff
@@ -774,6 +774,43 @@ struct TConfirmUnusedRemoteFilesResult {
     1: optional list<Types.TTabletId> confirmed_tablets
 }
 
+struct TPrivilegeParams {
+    1: required string priv_level
+    2: optional string ctl
+    3: optional string db
+    4: optional string tbl
```
Comment (xiaojunjie) [inline gensrc/thrift/FrontendService.thrift:781]: Only id but no tbl name in be ?

**Comment by yongjinhou** [comment]
run buildall

File: gensrc/thrift/FrontendService.thrift (hunk 2 above)
Comment (yongjinhou) [reply gensrc/thrift/FrontendService.thrift:790]: The Url carries table name

**Comment by yongjinhou** [comment]
run p0

**Comment by yongjinhou** [comment]
run compile

**Comment by yongjinhou** [comment]
run compile

**Comment by yongjinhou** [comment]
run buildall

**Comment by yongjinhou** [comment]
run clickbench

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**